static/app/views/starfish/views/appStartup/screenSummary/deviceClassBreakdownBarChart.tsx

```diff
@@ -9,40 +10,109 @@ import {
   getDurationUnit,
   tooltipFormatter,
 } from 'sentry/utils/discover/charts';
+import EventView from 'sentry/utils/discover/eventView';
 import {aggregateOutputType} from 'sentry/utils/discover/fields';
+import {DiscoverDatasets} from 'sentry/utils/discover/types';
 import {formatVersion} from 'sentry/utils/formatters';
+import {decodeScalar} from 'sentry/utils/queryString';
+import {MutableSearch} from 'sentry/utils/tokenizeSearch';
+import {useLocation} from 'sentry/utils/useLocation';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import {prepareQueryForLandingPage} from 'sentry/views/performance/data';
 import {LoadingScreen} from 'sentry/views/starfish/components/chart';
 import MiniChartPanel from 'sentry/views/starfish/components/miniChartPanel';
 import {useReleaseSelection} from 'sentry/views/starfish/queries/useReleases';
+import {SpanMetricsField} from 'sentry/views/starfish/types';
 import {formatVersionAndCenterTruncate} from 'sentry/views/starfish/utils/centerTruncate';
+import {STARFISH_CHART_INTERVAL_FIDELITY} from 'sentry/views/starfish/utils/constants';
+import {appendReleaseFilters} from 'sentry/views/starfish/utils/releaseComparison';
+import {YAxis, YAXIS_COLUMNS} from 'sentry/views/starfish/views/screens';
+import {useTableQuery} from 'sentry/views/starfish/views/screens/screensTable';
+import {transformDeviceClassEvents} from 'sentry/views/starfish/views/screens/utils';
 
+const YAXES = [YAxis.COLD_START, YAxis.WARM_START];
 const XAXIS_CATEGORIES = ['high', 'medium', 'low', 'Unknown'];
-const CHART_HEIGHT = 70;
 
 interface DeviceClassBreakdownBarChartProps {
-  isError: boolean;
-  isLoading: boolean;
-  title: string;
-  yAxis: string;
-  data?: Series[];
+  additionalFilters?: string[];
+  chartHeight?: number;
 }
 
 function DeviceClassBreakdownBarChart({
-  isLoading,
-  isError,
-  title,
-  data,
-  yAxis,
+  chartHeight,
+  additionalFilters,
 }: DeviceClassBreakdownBarChartProps) {
+  const pageFilter = usePageFilters();
+  const location = useLocation();
+  const {query: locationQuery} = location;
   const {
     primaryRelease,
     secondaryRelease,
     isLoading: isReleasesLoading,
   } = useReleaseSelection();
 
+  const startType = decodeScalar(location.query[SpanMetricsField.APP_START_TYPE]) ?? '';
+  const yAxis = YAXIS_COLUMNS[startType === 'cold' ? YAxis.COLD_START : YAxis.WARM_START];
```

Comment:
There's a constant I can replace `'cold'` with. I'll update after the first review pass